examples/cloud-operations/network-dashboard/main.tf

```diff
@@ -0,0 +1,165 @@
+  limit_l7_list            = tolist(var.limit_l7)
+  limit_l7_ppg             = join(",", local.limit_l7_ppg_list)
+  limit_l7_ppg_list        = tolist(var.limit_l7_ppg)
+  limit_subnets            = join(",", local.limit_subnets_list)
+  limit_subnets_list       = tolist(var.limit_subnets)
+  limit_vpc_peer           = join(",", local.limit_vpc_peer_list)
+  limit_vpc_peer_list      = tolist(var.limit_vpc_peer)
+}
+
+################################################
+# Monitoring project creation                  #
+################################################
+
+module "project-monitoring" {
+  source          = "github.com/GoogleCloudPlatform/cloud-foundation-fabric//modules/project?ref=v14.0.0"
+  name            = "monitoring"
+  parent          = "organizations/${var.organization_id}"
+  prefix          = var.prefix
+  billing_account = var.billing_account
+  services        = var.project_monitoring_services
+}
+
+################################################
+# Service account creation and IAM permissions #
+################################################
+
+module "service-account-function" {
+  source       = "github.com/GoogleCloudPlatform/cloud-foundation-fabric//modules/iam-service-account?ref=v14.0.0"
+  project_id   = module.project-monitoring.project_id
+  name         = "sa-dash"
+  generate_key = false
+
+  # Required IAM permissions for this service account are:
+  # 1) compute.networkViewer on projects to be monitored (I gave it at organization level for now for simplicity)
+  # 2) monitoring viewer on the projects to be monitored (I gave it at organization level for now for simplicity)
+  iam_organization_roles = {
+    "${var.organization_id}" = [
+      "roles/compute.networkViewer",
+      "roles/monitoring.viewer",
+    ]
+  }
+
+  iam_project_roles = {
+    "${module.project-monitoring.project_id}" = [
+      "roles/monitoring.metricWriter"
+    ]
+  }
+}
+
+################################################
+# Cloud Function configuration (& Scheduler)   #
+################################################
+
+module "pubsub" {
+  source     = "github.com/GoogleCloudPlatform/cloud-foundation-fabric//modules/pubsub?ref=v14.0.0"
+  project_id = module.project-monitoring.project_id
+  name       = "network-dashboard-pubsub"
+  subscriptions = {
+    "network-dashboard-pubsub-default" = null
+  }
+  # the Cloud Scheduler robot service account already has pubsub.topics.publish
+  # at the project level via roles/cloudscheduler.serviceAgent
+}
+
+resource "google_cloud_scheduler_job" "job" {
+  project   = module.project-monitoring.project_id
+  region    = var.region
+  name      = "network-dashboard-scheduler"
+  schedule  = var.schedule_cron
+  time_zone = "UTC"
+
+  pubsub_target {
+    topic_name = module.pubsub.topic.id
+    data       = base64encode("test")
+  }
+}
+
+# Random ID to re-deploy the Cloud Function  with every Terraform run
+resource "random_pet" "random" {
```

Comment:
I might be missing something but I don't think you need this. The CF module will zip all the source files and redploy if the md5 of the resulting file changes. Do you really need a random bucket id every run?